drush/cm_tools.drush.inc

```diff
@@ -0,0 +1,163 @@
+<?php
+
+/**
+ * Implementation of hook_drush_sql_sync_sanitize().
+ */
+function cm_tools_drush_sql_sync_sanitize() {
+  // Fetch list of all tables.
+  $all_tables = drush_sql_get_class()->listTables();
+
+
+  // User module.
+  $queries = array();
+
+  if (in_array('users', $all_tables, TRUE)) {
+    $queries[] = "UPDATE users SET name = CONCAT('user-' , uid) WHERE uid > 0";
+    $queries[] = "UPDATE users SET mail = CONCAT('user-' , uid, '@example.com'), init = CONCAT('user-' , uid, '@example.com') WHERE uid > 0";
+  }
+
+  if (!empty($queries)) {
+    $queries[] = '';
+    drush_sql_register_post_sync_op('user', dt('Sanitize user information.'), implode(';', $queries));
+  }
+
+
+  // Redirect module.
+  $queries = array();
+
+  if (in_array('redirect', $all_tables, TRUE)) {
+    // Redirects may contain personal data like usernames etc.
+    $queries[] = "DELETE FROM redirect WHERE redirect LIKE 'user/%'";
+  }
+
+  if (!empty($queries)) {
+    $queries[] = '';
+    drush_sql_register_post_sync_op('redirect', dt('Sanitize redirect to user paths.'), implode(';', $queries));
+  }
+
+
+  // Path module.
+  $queries = array();
+
+  if (in_array('url_alias', $all_tables, TRUE)) {
+    // URL aliases may contain personal data like usernames etc.
+    $queries[] = "DELETE FROM url_alias WHERE source LIKE 'user/%'";
+  }
+
+  if (!empty($queries)) {
+    $queries[] = '';
+    drush_sql_register_post_sync_op('path', dt('Sanitize URL aliases.'), implode(';', $queries));
+  }
+
+  // Comment module.
+  $queries = array();
+
+  if (in_array('comment', $all_tables, TRUE)) {
+    $queries[] = "UPDATE comment SET hostname = '192.0.2.1'";
+    $queries[] = "UPDATE comment SET name = sha(name) WHERE name <> ''";
+    $queries[] = "UPDATE comment SET mail = CONCAT(sha(mail), '@example.com') WHERE mail <> ''";
+    $queries[] = "UPDATE comment SET homepage = CONCAT(sha(homepage), '.example.com') WHERE homepage <> ''";
```

Comment:
Given that users often sign their names at the bottom of comments, is it worth anonymizing the body field for comments? The GDPR module uses the [PHP faker library](https://github.com/fzaninotto/Faker) like [this](https://cgit.drupalcode.org/gdpr/tree/modules/anonymizer/src/Plugin/Anonymizer/TextAnonymizer.php) which works pretty neatly. That might be a step beyond what you want to do, but the thought is here :-)

Reply:
Yeah, I guess the actual content of comments is rarely something we care about so replacing them with fake ones probably isn't a bad shout.

Reply:
Yeah, mebbe...tricky though!